alchemy/src/aws/scope-extensions.ts

```diff
@@ -0,0 +1,18 @@
+import type { AwsClientProps } from "./client-props.ts";
+
+/**
+ * AWS scope extensions - adds AWS credential support to scope options.
+ * This uses TypeScript module augmentation to extend the ProviderCredentials interface.
+ * Since ExtendedScopeOptions and RunOptions both extend ProviderCredentials,
+ * they automatically inherit these properties.
+ */
+declare module "../scope.ts" {
+  interface ProviderCredentials {
+    /**
+     * AWS credentials configuration for this scope.
+     * All AWS resources created within this scope will inherit these credentials
+     * unless overridden at the resource level.
+     */
+    aws?: AwsClientProps;
+  }
+}
```

Comment:
nit: you could move this augmentation to the `./client-props.ts` so that we never need to do any naked imports.